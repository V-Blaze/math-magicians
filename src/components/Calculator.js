import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
const Calculator = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleCalculation = (e) => {
    const buttonName = e.target.name;

    return setCalcData((obj) => calculate(obj, buttonName));
  };

  const { total, next, operation } = calcData;
  const output = `${total}${operation}${next}`.replace(/null/g, '').replace(/undefined/g, '');

  return (
    <>
      <div className="calculator">

        <input
          type="text"
          className="calculator-screen"
          id="calc-screen"
          value={output === '' ? '0' : output}
          disabled
        />

        <div className="calculator-keys">

          <button type="button" className="all-clear" name="AC" onClick={handleCalculation}>AC</button>
          <button type="button" className="Plus-minus" name="+/-" onClick={handleCalculation}>+/-</button>
          <button type="button" className="oprator" name="%" onClick={handleCalculation}>%</button>
          <button type="button" className="operator" name="+" onClick={handleCalculation}>+</button>

          <button type="button" className="number" name="7" onClick={handleCalculation}>7</button>
          <button type="button" className="number" name="8" onClick={handleCalculation}>8</button>
          <button type="button" className="number" name="9" onClick={handleCalculation}>9</button>
          <button type="button" className="operator" name="x" onClick={handleCalculation}>&times;</button>

          <button type="button" className="number" name="4" onClick={handleCalculation}>4</button>
          <button type="button" className="number" name="5" onClick={handleCalculation}>5</button>
          <button type="button" className="number" name="6" onClick={handleCalculation}>6</button>
          <button type="button" className="operator" name="-" onClick={handleCalculation}>-</button>

          <button type="button" className="number" name="1" onClick={handleCalculation}>1</button>
          <button type="button" className="number" name="2" onClick={handleCalculation}>2</button>
          <button type="button" className="number" name="3" onClick={handleCalculation}>3</button>
          <button type="button" className="operator" name="รท" onClick={handleCalculation}>&divide;</button>

          <button type="button" className="number zero" name="0" onClick={handleCalculation}>0</button>
          <button type="button" className="number" name="." onClick={handleCalculation}>.</button>
          <button type="button" className="operator" name="=" onClick={handleCalculation}>=</button>

        </div>

      </div>
    </>
  );
};

export default Calculator;
