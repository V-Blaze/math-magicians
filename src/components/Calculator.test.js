import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import Calculator from './Calculator';

describe('testing the calculator', () => {
  it('snapshots', () => {
    const tree = renderer
      .create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing the h1 inside the Calculator ', () => {
    render(<Calculator />);
    const h1Element = screen.getByText(/Lets do some math!/i);
    expect(h1Element).toBeInTheDocument();
  });
  it('testing the number (4) inside the Calculator ', () => {
    render(<Calculator />);
    const buttonNumberElement = screen.getByText(/4/i);
    expect(buttonNumberElement).toBeInTheDocument();
  });
  it('testing the (=) inside the Calculator ', () => {
    render(<Calculator />);
    const operatorElement = screen.getByText(/=/i);
    expect(operatorElement).toBeInTheDocument();
  });
});
