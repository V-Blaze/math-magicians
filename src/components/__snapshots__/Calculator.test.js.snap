// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the calculator snapshots 1`] = `
<div
  className="calculator-main"
>
  <div
    className="main-title-div"
  >
    <h1
      className="main-title"
    >
      Lets do some math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <input
      className="calculator-screen"
      disabled={true}
      id="calc-screen"
      type="text"
      value="0"
    />
    <div
      className="calculator-keys"
    >
      <button
        className="all-clear"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="Plus-minus"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="oprator"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="number"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="number"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="number"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        name="x"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
      <button
        className="number"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="number"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="number"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="number"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="number"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="number"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="number zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="number"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`testing the calculator testing the calculator 1`] = `
<div
  className="calculator-main"
>
  <div
    className="main-title-div"
  >
    <h1
      className="main-title"
    >
      Lets do some math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <input
      className="calculator-screen"
      disabled={true}
      id="calc-screen"
      type="text"
      value="0"
    />
    <div
      className="calculator-keys"
    >
      <button
        className="all-clear"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="Plus-minus"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="oprator"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="number"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="number"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="number"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        name="x"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
      <button
        className="number"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="number"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="number"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="number"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="number"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="number"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="number zero"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="number"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
